pkg/repo/fs.go

```diff
@@ -96,17 +96,16 @@ func (fsr *FsRepo) Init(c config.ReadWriter) error {
 		return err
 	}
 
-	// modifies the config to include keys for accessing repo paths if they are not set.
-	// This ensures either user provided paths are valid to default paths for the repo are set.
-	fsr.EnsureRepoPathsConfigured(c)
-
 	cfg, err := c.Current()
 	if err != nil {
 		return err
 	}
 
-	if err := initRepoFiles(cfg); err != nil {
-		return fmt.Errorf("failed to initialize repo: %w", err)
+	// in the event a user has provided a config without a path we need to set it here based on what the repo was
+	// initialized with.
+	if cfg.DataDir == "" {
```

Comment:
Issue to move away from this here: https://github.com/bacalhau-project/bacalhau/issues/4337